### ISSUE TITLE
18% performance regression on x86 introduced by Update CodeGen_PTX_Dev to use new PassManager (#6718)

### ISSUE THREAD

**Issue by froody** [halide/Halide#6783]
460c77eca Update CodeGen_PTX_Dev to use new PassManager (#6718)

I updated to main (ad1e7f6a6) and noticed that my pipeline took was much slower compared to Halide-v13.0.4. I ran git bisect and identified the above commit as the cause of a 18% regression in my pipeline on x86, i.e. my benchmark results:

Before: 0.5851s
After: 0.6914s

I'm not sure if I can share the details of my pipeline but if you have trouble reproducing the problem then let me know and I'll see what I can do.

llvm-version: llvmorg-13.0.0
cmake invocation: 
```
cmake -G Ninja \
                 -DCMAKE_BUILD_TYPE=Release \
                 -DWITH_PYTHON_BINDINGS=ON \
                 -DPYTHON_EXECUTABLE=$CONDA_ENV/bin/python3.9 \
                 -DPython3_ROOT=$CONDA_ENV \
                 -DLLVM_TARGETS_TO_BUILD="host;JSBackend;WebAssembly" \
                 -DLLVM_INCLUDE_EXAMPLES=OFF -DLLVM_INCLUDE_TESTS=OFF -DCLANG_INCLUDE_TESTS=OFF \
                 ..
```
CPU: AMD EPYC 7742

**Comment by abadams** [comment]
Thanks for the regression report. Which LLVM version are you compiling against? Or are you using a prebuilt binary?

**Comment by abadams** [comment]
Hopefully fixed by #6785
